### DISCUSSION TITLE
How do you think about use libfork for DAG tasks

### DISCUSSION THREAD

**Discussion by Frandy** [ConorWilliams/libfork#24] [General]
Libfork is great for parallel-for or nested job. How about DAG tasks ? We can maintain task dependency and call libfork for parallel. It is  one kind of nested jobs, but the difference is it  does not need to wait for child to finish. For DAG task parallel system, we can find  [taskflow](https://github.com/taskflow/taskflow) and [scheduling](https://github.com/dpuyda/scheduling). For very large DAG, will it cause problem with too deep nest ?

**Comment by ConorWilliams** [comment]
Libfork implicitly models all tasks as a DAG (this is what enables so much of the theoretical analysis); libfork can handle arbitrarily deeply nested DAGs with its dynamic stacks. Waiting for child tasks to complete is one of the key constraints of _strict_ fork-join parallelism, that allows the strong memory/time bounds, so we will unfortunetly not be loosening this constraint.
